langchain/src/document_loaders/fs/directory.ts

```diff
@@ -19,7 +19,7 @@ export class DirectoryLoader extends BaseDocumentLoader {
   constructor(
     public directoryPath: string,
     public loaders: {
-      [extension: string]: (filePath: string) => BaseDocumentLoader;
+      [extension: `.${string}`]: (filePath: string) => BaseDocumentLoader;
```

Comment:
Since we're always expecting the extension to start with a `.`, this should help TS users to have better DX. wdyt?